client/src/game/gameObjects/moveableGameObject.ts

```diff
@@ -25,36 +25,35 @@ class MoveableGameObject extends GameObject implements IMoveableGameObject {
 				}
 			}
 		}
-
 		super.update(time, delta)
 		// updateData('position')
 	}
 
 	private move(source: ICordinates, destination: ICordinates, delta: number): ICordinates {
-		return {
-			x: this.getDistanceToMove(source.x, destination.x, delta),
-			y: this.getDistanceToMove(source.y, destination.y, delta),
-			z: this.getDistanceToMove(source.z, destination.z, delta)
+		const direction = {
+			x: destination.x - source.x,
+			y: destination.y - source.y,
+			z: destination.z - source.z
 		}
-	}
-
-	private getDistanceToMove = (source: number, destination: number, delta: number): number => {
-		let newPostion: number = source
-		let moveDistance = delta * this.movementSpeed
-		const diff = Math.abs(destination - source)
-		if (diff === 0) {
-			return destination
+		const posToDesLength = Math.sqrt(direction.x * direction.x + direction.y * direction.y + direction.z * direction.z)
+		if (posToDesLength === 0) {
+			return source
 		}
-		if (diff < moveDistance) {
-			moveDistance = diff
+		const velocity = {
```

Comment:
Apologies for my ignorance, but what is delta in this context?

Reply:
This could be named better. Delta here is the name of "Time elapsed since the last tick", where a tick is run of the gameloop.

You can see an example of it on the server here:
TimeResource::elapsed_seconds

Reply:
Somehow I messed up the lines, which I referred to in my review comments. My apologies. Gotcha tho. So can we change the name? So noobs like me can also read what's going on

Reply:
Is parameter name length also a performance consideration ..?

Reply:
No, variable and parameter names has no effect on performance.... unless you make them super long and then start to do object indexing:
var key = "mySuperLongVariableNameThatWillMeanThatWeNeedALongerStringToIndexIt";
obj[key]

Reply:
Good, thank you for that clarification 👍 In that case I'd suggest a rename for more specificity

Reply:
Renamed it for this function. I would suggest to make a new pull request if it needs to be a client-wide change. I feel it is outside the scope of the PR.